crypto/src/merkle/mod.rs

```diff
@@ -214,9 +214,12 @@ impl<H: Hasher> MerkleTree<H> {
         if indexes.len() > proofs::MAX_PATHS {
             return Err(MerkleTreeError::TooManyLeafIndexes(proofs::MAX_PATHS, indexes.len()));
         }
+        let mut indexes = indexes.to_vec();
+        indexes.sort_unstable();
+        indexes.dedup();
```

Comment:
I'm not sure this part is needed as both `map_indexes()` and `normalize_indexes()` should naturally de-duplicate the data.

I think here we should add the following:
- Update comments to this function to describe the new behavior and remove the obsolete error condition.
- Similarly update comments for [verify_batch()](https://github.com/facebook/winterfell/blob/main/crypto/src/merkle/mod.rs#L313), [BatchMerkleProof::get_root()](https://github.com/facebook/winterfell/blob/main/crypto/src/merkle/proofs.rs#L123), and maybe other function.
- Add comments to `map_indexes()` and `normalize_indexes()`.